pandas/tests/io/parser/common/test_chunksize.py

```diff
@@ -181,8 +181,14 @@ def test_warn_if_chunks_have_mismatched_type(all_parsers):
     if parser.engine == "c" and parser.low_memory:
         warning_type = DtypeWarning
 
-    with tm.assert_produces_warning(warning_type):
+    with tm.assert_produces_warning(warning_type) as record:
         df = parser.read_csv(StringIO(data))
+    if record:
+        assert (
+            str(record[0].message)
+            == "Columns (0) have mixed types. Specify dtype option on import or "
```

Comment:
The string is quite long, so it doesn't look particularly pretty in here, but I think it's okay-ish.

Reply:
usually we split this into two lines

```
expected = "Columns (0) ..."
assert str(record[0].message) == expected
```